fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/MessageHelpers.java

```diff
@@ -560,5 +562,10 @@ public CoercionTrieNode(@Nullable final BiFunction<Descriptors.Descriptor, Objec
         public CoercionTrieNode getThis() {
             return this;
         }
+
+        @Override
+        public int planHash(@Nonnull final PlanHashKind hashKind) {
+            return PlanHashable.objectsPlanHash(hashKind, getChildrenMap() == null ? ImmutableSet.of() : getChildrenMap().keySet(), getChildren());
```

Comment:
is the `keySet` stable? I think until we fix a problem with plan hashable that treats sets as an iterable you will have to hashCode the set here directly or you fix the set problem in planhashable.